test/language/class/class-strict-mode.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class strict mode
+---*/
+class C {}
+
+with ({a: 1}) {
+  assert.sameValue(a, 1);
+}
+
+assert.throws(SyntaxError,
+             'class C extends function B() { with ({}); return B; }() {}');
```

Comment:
Yeah, good catch. Funny thing though: 


```js
assert.throws(SyntaxError, function() {
  class C extends (function B() { with ({}); return B; }()) {}
});
```

Fails with: 

```
$ test262 --command "v8 --harmony-classes --harmony-computed-property-names --harmony-sloppy" class-strict-mode
=== language/class/class-strict-mode failed in non-strict mode ===
--- output ---
 /var/folders/jl/3l91m2l97xjbh10v99_xmwmh0000gn/T/test262-hMYUkc.js:130: SyntaxError: Strict mode code may not include a with statement
  class C extends (function B() { with ({}); return B; }()) {}
                                  ^^^^
SyntaxError: Strict mode code may not include a with statement===
```

How is that not caught?